spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala

```diff
@@ -55,7 +55,12 @@ case class Etl(args: Seq[String], @transient modules: Seq[TypedModule] = Etl.def
   def scheme: Either[LayoutScheme, LayoutDefinition] = {
     if (conf.layoutScheme.isDefined) {
       val scheme = conf.layoutScheme()(conf.crs(), conf.tileSize())
-      logger.info(scheme.toString)
+      if (conf.maxZoom.isDefined) {
+        scheme match {
+          case  zoomedLayoutScheme: ZoomedLayoutScheme => logger.info("maxZoom and layout scheme compatible")
+          case _ => throw new RuntimeException("maxZoom option should only be used with tms layout scheme")
+        }
+      }
```

Comment:
Ah, that's a good point. That makes it more difficult than it should be. I think rather than trying to get fancy with config parsing lets stick with your original approach.